pkg/apis/pipeline/v1/pipeline_types.go

```diff
@@ -478,9 +478,15 @@ func (pt PipelineTask) Validate(ctx context.Context) (errs *apis.FieldError) {
 	errs = errs.Also(pt.validateRefOrSpec())
 
 	errs = errs.Also(pt.validateEmbeddedOrType())
-
+	notCustomTaskKinds := map[TaskKind]bool{
+		"":                 true,
+		NamespacedTaskKind: true,
+		"ClusterTask":      true,
+	}
 	// Pipeline task having taskRef/taskSpec with APIVersion is classified as custom task
 	switch {
+	case pt.TaskRef != nil && !notCustomTaskKinds[pt.TaskRef.Kind]:
+		errs = errs.Also(pt.validateCustomTask())
 	case pt.TaskRef != nil && pt.TaskRef.APIVersion != "":
 		errs = errs.Also(pt.validateCustomTask())
 	case pt.TaskSpec != nil && pt.TaskSpec.APIVersion != "":
```

Comment:
It looks like this change will only apply to taskRefs, and not taskSpecs. It also looks like it will only apply to taskRefs in pipeline tasks, and not taskrefs in taskruns. What's the reason for this?

Reply:
I think we probably miss custom task validation for taskSpecs and taskrefs in taskruns. That may be an issue we need to address. I don't have too much context/background on why we only have the "validateCustomTask" for pipelineTask. 🙃

Reply:
Sorry I'm not sure what needs to be modified for `isCustomTask`, isCustomTask is used to tell if it is custom task, it requires both of the apiversion and kind. But here we're validating it, so if either is empty we should return error.

Reply:
Thanks Yongxuan! would you mind testing whether validation is applied for custom tasks in taskrun refs and creating an issue for that?

Re: isCustomTask, I think returning an error is probably the right approach even though we'll have to add the extra return value

Reply:
yes, taskrun ref has the same issue, this is a tracking issue: https://github.com/tektoncd/pipeline/issues/6557
Thanks!

Reply:
Thanks! would you mind just noting in the commit message/PR description that taskrun taskrefs will be handled separately, and linking to the issue in the PR description?